### PR TITLE
Add co2 savings

### PR DESCRIPTION
**Pull request by yerbol-akhmetov**
Good day. I propose to add a script for calculating CO2 emission savings for each scenario in tCO2_eq and equivalent natural gas volume in m3.

### CODE REVIEW

File: plots/plot_co2_level.py [hunk 1]
```diff
@@ -335,6 +338,8 @@ def fill_table_df(df, planning_horizon, scenarios, values):
         if not co2_df.empty:
             plot_co2_balance(co2_df, clusters, planning_horizon)
             table_co2_df = fill_table_df(table_co2_df, planning_horizon, scenarios, co2_df)
+            co2_emission = (co2_df.abs().sum() / 2).to_frame().T
```
Comment (martacki) [inline plots/plot_co2_level.py:341]: why divided by 2?

File: plots/plot_co2_level.py [hunk 2]
```diff
@@ -361,6 +366,13 @@ def fill_table_df(df, planning_horizon, scenarios, values):
             plot_co2_balance(co2_BAU, clusters, BAU_horizon, plot_width=1.5)
             table_co2_df = fill_table_df(table_co2_df, BAU_horizon, {"BAU":"BAU"}, co2_BAU)
             table_co2_df = table_co2_df.fillna(0)
+            co2_emission_BAU = (co2_BAU.abs().sum() / 2)[0]
```
Comment (martacki) [inline plots/plot_co2_level.py:369]: also, why need to be divided by 2?

File: plots/plot_electricity_for_heat.py [hunk 3]
```diff
@@ -76,15 +77,15 @@ def plot_elec_consumption_for_heat(dict_elec):
...
+    if "BAU" in dict_elec.keys():
         axes = axes * 4
     i=0
     for name, elec_demand_f_heating in dict_elec.items():
         ax = axes[i]
         ax.set_facecolor("whitesmoke")
 
         print("Hard coded coordinates on x-axis, selected for 3H granularity")
-        where = [56, 224]
+        where = [359, 527]
```
Comment (martacki) [inline plots/plot_electricity_for_heat.py:88]: I theory we could also explore if 2 weeks for example would be better to represent, but we can try it ad hoc outside of this PR

File: plots/plot_electricity_for_heat.py [hunk 4]
```diff
@@ -95,14 +96,19 @@ def plot_elec_consumption_for_heat(dict_elec):
         )
 
         ax.set_xlabel("", fontsize=12)
-        ax.set_ylim([0,1200])
-        ax.set_yticks(np.arange(0, 1200, 500))
+        ax.set_ylim([1,2000])
```
Comment (martacki) [inline plots/plot_electricity_for_heat.py:99]: Are costs really so high? :fearful:

File: scripts/set_capacities.py [hunk 5]
```diff
@@ -87,6 +87,7 @@ def set_optimal_capacities(solved_network, unsolved_network):
 
     # set p_nom_max as max(p_nom_max, p_nom_min) for retrofitting
     retro_idx = unsolved_network.generators.query("carrier == 'retrofitting'").index
+    unsolved_network.generators.loc[retro_idx, ["p_nom_min", "p_nom"]] = solved_network.generators.loc[retro_idx, "p_nom_opt"]
```
Comment (martacki) [inline scripts/set_capacities.py:90]: with this, we need to set `p_nom_extendable` to false in certain simulations, e.g. after `heat_sink_T` was updated and for flexible-moderate. Did you test that it works for all scenarios?

**Review by martacki — APPROVED** [review]
LGTM!

File: scripts/set_capacities.py (hunk 5 above)
Comment (yerbol-akhmetov) [reply scripts/set_capacities.py:90]: I have tested in flexible and retro_tes scenarios with all horizons. It works even with `p_nom_extendable = false`.

**Pull request merged**